### PR TITLE
btrfs-progs: fix musl build

### PR DESCRIPTION
**Pull request by kampka**
###### Motivation for this change

Building with `musl` fails because `systemd` cannot be built with anything but `glibc` / GNU.
Reproducible with
```
nix build nixpkgs#pkgsMusl.btrfsProgs
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/filesystems/btrfs-progs/default.nix [hunk 1]
```diff
@@ -31,12 +32,21 @@ stdenv.mkDerivation rec {
...
+  makeFlags = lib.optionals stdenv.hostPlatform.isGnu [ "udevruledir=$(out)/lib/udev/rules.d" ];
 
   enableParallelBuilding = true;
 
+  passthru.tests = {
+    simple-filesystem = runCommand "btrfs-progs-create-fs" {} ''
+      mkdir -p $out
+      truncate -s110M $out/disc
+      ${btrfsProgs}/bin/mkfs.btrfs $out/disc | tee $out/success
+      ${btrfsProgs}/bin/fsck.btrfs $out/disc | tee $out/success
```
Comment (symphorien) [inline pkgs/tools/filesystems/btrfs-progs/default.nix:46]: fsck.btrfs is a noop, see the manpage
better trying btrfs check maybe ?

File: pkgs/tools/filesystems/btrfs-progs/default.nix [hunk 2]
```diff
@@ -31,12 +32,21 @@ stdenv.mkDerivation rec {
...
 
-  makeFlags = [ "udevruledir=$(out)/lib/udev/rules.d" ];
+  makeFlags = lib.optionals stdenv.hostPlatform.isGnu [ "udevruledir=$(out)/lib/udev/rules.d" ];
 
   enableParallelBuilding = true;
 
+  passthru.tests = {
+    simple-filesystem = runCommand "btrfs-progs-create-fs" {} ''
+      mkdir -p $out
+      truncate -s110M $out/disc
```
Comment (symphorien) [inline pkgs/tools/filesystems/btrfs-progs/default.nix:44]: you should probably `set -o pipefail`

Comment (kampka) [reply pkgs/tools/filesystems/btrfs-progs/default.nix:44]: I believer that is already set here by the generic builder: https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/setup.sh#L2

**Comment by kampka** [comment]
@symphorien @7c6f434c Thanks for the review :)

**Pull request merged**